mava/components/executing/action_selection.py

```diff
@@ -112,31 +106,119 @@ def on_execution_select_actions(self, executor: SystemExecutor) -> None:
         )
 
-    # NB: Not currently used. TODO Deprecate in future.
-    def on_execution_select_action_compute(self, executor: SystemExecutor) -> None:
-        """Select action for a single agent and save in store.
+    def on_execution_init_end(self, executor: SystemExecutor) -> None:
+        """Create function that is used to select actions.
 
         Args:
-            executor: SystemExecutor.
+            executor : SystemExecutor.
 
         Returns:
             None.
         """
+        networks = executor.store.networks
+        agent_net_keys = executor.store.agent_net_keys
+
+        def select_action(
+            observation: NestedArray,
+            current_params: NestedArray,
+            network: Any,
+            base_key: networks_lib.PRNGKey,
+        ) -> Tuple[NestedArray, NestedArray, networks_lib.PRNGKey]:
+            """Action selection across a single agent.
+
+            Args:
+                observation : The observation for the current agent.
+                current_params : The parameters for current agent's network.
+                network : The network object used by the current agent.
+                key : A JAX prng key.
+
+            Returns:
+                action info, policy info and new key.
+            """
+            observation_data = utils.add_batch_dim(observation.observation)
+            # We use the action_key immediately and keep the new key for future splits.
+            base_key, action_key = jax.random.split(base_key)
+            action_info, policy_info = network.get_action(
+                observations=observation_data,
+                params=current_params,
+                base_key=action_key,
+                mask=utils.add_batch_dim(observation.legal_actions),
+            )
+            return action_info, policy_info, base_key
+
+        def select_actions(
+            observations: Dict[str, NestedArray],
+            current_params: Dict[str, NestedArray],
+            base_key: networks_lib.PRNGKey,
+        ) -> Tuple[
+            Dict[str, NestedArray], Dict[str, NestedArray], networks_lib.PRNGKey
+        ]:
+            """Select actions across all agents - this is jitted below.
 
-        agent = executor.store.agent  # Set by Executor
-        network = executor.store.networks[executor.store.agent_net_keys[agent]]
+            Args:
+                observations : The observations for all the agents.
+                current_params : The parameters for all the agents.
+                base_key : A JAX prng_key.
 
-        # executor.store.observation set by Executor
-        observation = utils.add_batch_dim(executor.store.observation.observation)
-        rng_key, executor.store.base_key = jax.random.split(executor.store.base_key)
+            Returns:
+                action info, policy info and new prng key.
+            """
+            actions_info, policies_info = {}, {}
+            # TODO Look at tree mapping this forloop.
```

Comment:
Are you going to work on this TODO in this PR or in the future?

Reply:
This TODO has related to a copy of the feedforward TODO. I think we can address this in a future PR.